source/slang/slang-preprocessor.cpp

```diff
@@ -2403,97 +3508,111 @@ static void HandleDirective(PreprocessorDirectiveContext* context)
     expectEndOfDirective(context);
 }
 
+void Preprocessor::popInputFile()
+{
+    auto inputFile = m_currentInputFile;
+    SLANG_ASSERT(inputFile);
+
+    // We expect the file to be at its end, so that the
+    // next token read would be an end-of-file token.
+    //
+    auto expansionStream = inputFile->getExpansionStream();
+    Token eofToken = expansionStream->peekRawToken();
+    SLANG_ASSERT(eofToken.type == TokenType::EndOfFile);
+
+    // If there are any open preprocessor conditionals in the file, then
```

Comment:
I don't think C/C++ make this distinction, but think it's fair enough for our impl to.

Reply:
[MSDN}(https://docs.microsoft.com/en-us/cpp/preprocessor/hash-if-hash-elif-hash-else-and-hash-endif-directives-c-cpp?view=msvc-160#remarks) seems to indicate that C/C++ cares about this:

> All conditional-compilation directives, such as #if and #ifdef, must match a closing #endif directive before the end of file. Otherwise, an error message is generated. When conditional-compilation directives are contained in include files, they must satisfy the same conditions: There must be no unmatched conditional-compilation directives at the end of the include file.